plutus-core/plutus-core/src/PlutusCore/Subst.hs

```diff
@@ -145,6 +148,64 @@ termSubstFreeNames
     -> Term tyname name uni fun ann
 termSubstFreeNames = purely termSubstFreeNamesA
 
+-- | Substitute a closed 'Type' for a type variable in a 'Type'. Does not descend under binders that
+-- bind the same variable as the one we're substituting for. The resulting 'Term' may and likely
```

Comment:
> Is this behaviour... weird? 

No? I thought it was the perfectly reasonable behavior. As in, the function does exactly the right thing. Do you suspect something?

Reply:
I was confused. Of course this means the same as "only substitute free names", which is perfectly reasonable. Maybe write that instead?

Reply:
Not "names" -- "name". Changed the comment to

```
-- | Substitute the given closed 'Type' for the given type variable in the given 'Type'. Does not
-- descend under binders that bind the same variable as the one we're substituting for (since from
-- there that variable is no longer free). The resulting 'Term' may and likely will not satisfy
-- global uniqueness.
```